numpy/lib/function_base.py

```diff
@@ -4326,6 +4327,69 @@ def meshgrid(*xi, **kwargs):
             return np.broadcast_arrays(*output)
 
 
+def meshgridify(*xi, **kwargs):
+    """
+    Converts tabular data to meshgrid format.
+
+    Returns array of function values `f` fitting to meshgrid arrays created
+    from `xi`. If no `f` is given, only a meshgrid from `xi` is returned.
+
+    Parameters
+    ----------
+    x1, x2,..., xn : array_like
+        1-D arrays representing the coordinates of a grid.
+    f : array-like, optional
+        Function values for coordinate tuples in `xi`.
+
+    Returns
+    -------
+    X1, X2,..., Xn, Z : ndarray
+        Returns meshgrid arrays X1, X2, ..., Xn with dimensions n, created from
+        the coordinate vectors `x1`, `x2`, ..., `xn`. When `f` is given, an
+        array fitting to the coordinates given in X1, X2, ..., Xn is also
+        returned.
+
+    See Also
+    --------
+    np.meshgrid : Construct a multi-dimensional "meshgrid".
+
+    Examples
+    --------
+    If there is no function to apply to a meshgrid to obtain the according
+    function values, but only a table of coordinate-value pairs is given, the
+    conventional `meshgrid` function is not applicable and meshgridify is very
+    useful.
+
+    >>> [X, Y, Z] = meshgridify([1, 1, 2, 2], [3, 4, 3, 4], f=[0, 1, 2, 3])
```

Comment:
You can drop those square bracket, almost looks like Matlab (TM) or something ;p